utils/prices.py

```diff
@@ -72,11 +73,22 @@ def get_sol_price() -> float:
         except Exception as e:
             LOG.warning(f"Get error when try to get SOL price from: {network}: {e}")
             time.sleep(5)
-    else:
-        raise AssertionError("Can't get SOL price for all networks")
     return result
 
 
+def get_sol_price_with_retry(timeout: int = 120) -> float:
+    price = get_sol_price()
+    started = time.time()
+    while price is None and (time.time() - started) < timeout:
+        print("Can't get SOL price")
+        time.sleep(3)
+        price = get_sol_price()
+        return price
+    if price is not None:
+        return price
+    raise TimeoutError("Failed to get SOL price within the timeout period.")
+
+
```

Comment:
we have a wait_condition method (utils/helpers.py), why don't you use it here?

Reply:
because wait_condition doesn't return the result and if i'll try to get sol price again after wait_condition i can get an error